src/AbstractAlgebra.jl

```diff
@@ -1192,12 +1192,12 @@ export Generic
 #
 ###############################################################################
 
-getindex(R::NCRing, s::Union{String, Char, Symbol}) = polynomial_ring(R, s)
-getindex(R::NCRing, s::Union{String, Char, Symbol}, ss::Union{String, Char}...) =
+getindex(R::NCRing, s::Union{Symbol, AbstractString, Char}) = polynomial_ring(R, s)
+getindex(R::NCRing, s::Union{Symbol, AbstractString, Char}, ss::Union{Symbol, AbstractString, Char}...) =
```

Comment:
This is *not* a problem of this PR, but just to say: this code allows writing `R[:x, "y"]` if I understand right. I think we probably want instead
```suggestion
getindex(R::NCRing, s::T, ss::T...) where T <: Union{Symbol, AbstractString, Char} =
```

Reply:
Well, I feel generally inclined, to allow things, if it does cost me nothing. If people want to write `R[:x, 'y']`, I let them. It looks strange, but it may arise in some context. Philosophies may differ.

Reply:
Actually yesterday I thought the code means that the first argument can be different, but the 2nd, 3rd, 4th, ... all have to have the same type, and that struck me as weird and undesirable. But looking again it seems all combinations are possible, so I withdraw my objections

Note that right now this feature doesn't work anyway but I expect your PR fixes that:
```
julia> ZZ['x',"y","z"]
ERROR: MethodError: no method matching polynomial_ring(::ZZRing, ::Vector{Any})
```